### ISSUE TITLE
local/root deps are not handled correctly when using paths outside referring project

### ISSUE THREAD

**Issue by cursive-ide** [cursive-ide/cursive#2829]
When a project uses a `:local/root` dep like `../some-project`, the deps are not handled correctly, leading to exceptions like:
```
java.lang.NullPointerException: Cannot invoke "java.io.File.getName()" because "next_dir" is null
	at cursive.deps.sync$with_unique_module_names$fn__18783.invoke(sync.clj:256)
	at clojure.lang.PersistentHashMap$NodeSeq.kvreduce(PersistentHashMap.java:1307)
	at clojure.lang.PersistentHashMap$BitmapIndexedNode.kvreduce(PersistentHashMap.java:802)
	at clojure.lang.PersistentHashMap.kvreduce(PersistentHashMap.java:236)
	at clojure.core$fn__8525.invokeStatic(core.clj:6908)
	at clojure.core$fn__8525.invoke(core.clj:6888)
	at clojure.core.protocols$fn__8257$G__8252__8266.invoke(protocols.clj:175)
	at clojure.core$reduce_kv.invokeStatic(core.clj:6919)
	at clojure.core$reduce_kv.invoke(core.clj:6910)
	at cursive.deps.sync$with_unique_module_names.invokeStatic(sync.clj:248)
```
and:
```
java.lang.IllegalStateException: Source folder file:///Users/onetom/github.com/thheller/shadow-cljs/../shadow-css/src/dev must be under content entry file:///Users/onetom/github.com/thheller/shadow-css
```

**Comment by cursive-ide** [comment]
Fix released in 1.13.1-eap2 on 2023-08-07.

**Comment by cursive-ide** [comment]
Fix released in 1.13.1 on 2023-12-07.
